File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    cnt = 0
    for i in range(0, len(s), 1):
        str_s = s[i]
        if str_s == '(':
            cnt = cnt + 1
        else:
            cnt = cnt - 1
        if cnt == -1:
            break
    if cnt != 0:
        answer = False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        
        for (char i : s.toCharArray()){
            if ('(' == i) {
                stack.push(i);
            } else if (')' == i) {
                if (stack.isEmpty()) {
                    return false;
                } else {
                    stack.pop();
                }
            }
        }
        answer = stack.isEmpty();
        return answer;
    }
}